packages/gasket-plugin-command/lib/utils/process-command.js

```diff
@@ -0,0 +1,50 @@
+import { Command } from 'commander';
+import { processArgs } from './process-args.js';
+import { processOptions } from './process-options.js';
+import { createOption } from './create-option.js';
+const program = new Command();
+
+/**
+ * isValidCommand - Validates the command configuration
+ * @param {import('../index.d.ts').GasketCommandDefinition} command The command configuration
+ * @returns {boolean} True if valid, false otherwise
+ */
+function isValidCommand(command) {
+  const keys = Object.keys(command);
+  return keys.length &&
+    keys.length <= 7 &&
+    command &&
+    command.id &&
+    command.description &&
+    command.action &&
+    typeof command.action === 'function';
+}
+
+/**
+ * processCommand - Process the command configuration
+ * @type {import('../index.d.ts').processCommand}
+ */
+export function processCommand(command) {
+  if (!isValidCommand(command)) throw new Error('Invalid command configuration');
+  const { id, description, action, args, options, hidden = false, default: isDefault = false } = command;
+  const cmd = program
+    .command(id)
+    .description(description)
+    .action(action);
+
+  if (args) {
+    const cmdArgs = processArgs(args);
+    // @ts-expect-error
```

Comment:
Hm, I'm not sure that `Array<string | boolean>` is a useful return type for `processArgs`; you should use a type where you don't have to expect a TypeScript error.

Reply:
That is the accurate representation of the return type. Any suggestions on the typing?

Reply:
Quite frankly that error is not helpful whatsoever. I've spent a lot of time trying to find a solution to no avail hence the `@ts-expect-error`.

Reply:
You could be very specific and use `Parameters<Command['argument']>` if you want to explicitly make a type that is compatible with the parameters of a function.